### ISSUE TITLE
Unnecessary Default 10000 fee for MakePayment function

### ISSUE THREAD

**Issue by Can-Sahin** [blockchain/service-my-wallet-v3#351]
Using Service v0.26.0

Why is there a default fixed 10000 fee addition in `makePayment` function in case options are empty 

File `api.js `
``` javascript
MerchantAPI.prototype.makePayment = function (guid, options) {
    if (!isNaN(options.fee_per_byte)) {
        if (options.fee_per_byte < 50) warning = warnings.LOW_FEE_PER_BYTE
        payment.updateFeePerKb(options.fee_per_byte)
      } else if (!isNaN(options.fee)) {
        warning = warnings.STATIC_FEE_AMOUNT
        payment.fee(options.fee)
      } else {
        warning = warnings.USING_DEFAULT_FEE
        payment.fee(10000) // WHATS THE REASON OF THIS FALLBACK?
      }
```

Because when fee is unset, `blockchain-wallet-client` automatically sets fee per byte dynamically from its api [API](https://api.blockchain.info/mempool/fees) for a **regular** speed which is much more preferable.

Whats the point of this ? This forces developer to calculate fee_per_byte beforehand where not calculating could easily lead to a **regular** transaction. If there is no harm deleting it I would do a PR.

**Comment by Can-Sahin** [comment]
Yes I was thinking what you have said. I can suggest something like (dirty)

Before creating payment get fees from the API. Then add another IF condition to check fee_per_byte
(you need to change the type of fee_per_byte option at server.js. Its not only number anymore)
``` javascript
MerchantAPI.prototype.makePayment = function (guid, options) {
  return this.getWallet(guid, options)
    .then(requireSecondPassword(options))
    // Get fees from API
    .then(function (wallet) {
      return Promise.resolve(
        Blockchain.API.getFees().then(fees => {
          wallet.dynamicFees = fees
          return wallet
        }).catch(err => wallet)
      )
    })
    .then(function (wallet) {
      var from = isNaN(options.from)
        ? options.from : parseInt(options.from)

      var payment = wallet.createPayment()
        .to(options.to)
        .amount(options.amount)
        .from(from)

      var warning
      var dynamicFee = wallet.dynamicFees && wallet.dynamicFees[options.fee_per_byte]
      if (wallet.dynamicFees && isNaN(options.fee_per_byte) && !isNaN(dynamicFee)) {
          payment.updateFeePerKb(fee)
      } else if (!isNaN(options.fee_per_byte)) {
        if (options.fee_per_byte < 50) warning = warnings.LOW_FEE_PER_BYTE
        payment.updateFeePerKb(options.fee_per_byte)
      } else if (!isNaN(options.fee)) {
        warning = warnings.STATIC_FEE_AMOUNT
        payment.fee(options.fee)
      } else {
        warning = warnings.USING_DEFAULT_FEE
        payment.fee(10000)
      }
```

But I have another question. Is there a way to calculate transaction fee (in BTC) using 'service-my-wallet' ? I know fee depends on the transaction size and size depends on the inputs/outputs. So it isn`t straight forward. Now I can only get it by creating a `payment` without publishing it (Modified the code myself). Payment objects contains `finalFee` property which I need to show to my customer. Would you considering adding another interface to do this? Just like the makePayment function but without publishing it, so that payment properties are accessible beforehand

**Comment by jtormey** [comment]
Thanks for the example, will work on adding that this week!

It is inconvenient not to know the final fee until the transaction is published. Maybe there could be an option to perform a dry run, which makes a payment but does not publish. The dry run call could include a token that you pass to a second request when you want to continue with the payment.

**Comment by Can-Sahin** [comment]
Ok. It would be nice. I am already doing it now. I created my self a special condition if the payment 'to' is empty then I just build the payment and return it.  For dynamic fees or for this I can do a pull request if that works as well let me know.

**Comment by rockyxcoded** [comment]
Has this issue been fixed?

**Comment by prasitchai1991** [comment]
เมื่อ พ. 3 ก.ค. 2019 เวลา 09:16 Fulfil Nelson <notifications@github.com>
เขียนว่า:

**Comment by AEmad01** [comment]
Is this done yet? the dry run thing.
